src/porepy/models/verification_setups/terzaghi.py

```diff
@@ -0,0 +1,819 @@
+    """Method that non-dimensionalises pressure. The method is provided by the mixin
+    class :class:`SetupUtilities`.
+
+    """
+
+    exact_sol: ExactSolution
+    """Exact solution object."""
+
+    results: list[SaveData]
+    """List of :class:`SaveData` objects containing the results of the verification."""
+
+    numerical_consolidation_degree: Callable[[np.ndarray, np.ndarray], number]
+    """Method that computes the numerical degree of consolidation. The method is
+    provided by the mixin class :class:`SetupUtilities`.
+
+    """
+
+    relative_l2_error: Callable[[grid, np.ndarray, np.ndarray, bool, bool], number]
+    """Method that computes the discrete relative L2-error. The method is provided by
+    the mixin class:class:`porepy.models.verification_setups.VerificationUtils`.
+
+    """
+
+    def save_data(self) -> None:
+        """Save data to the `results` list.
+
+        Note:
+            Data will be appended to the ``results`` list only if the current time
+            matches a time from ``self.params["stored_times"]``. By default,
+            only the data for the final simulation time is stored.
+
+        """
+        t = self.time_manager.time
+        tf = self.time_manager.time_final
+        if any(np.isclose(t, self.params.get("stored_times", [tf]))):
+            collected_data = self._collect_data()
+            self.results.append(collected_data)
+
+    def _collect_data(self) -> SaveData:
+        """Collect data for the current simulation time.
+
+        Returns:
+            SaveData object containing the results of the verification.
+
+        """
+
+        # Retrieve data from setup
+        sd = self.mdg.subdomains()[0]
+        data = self.mdg.subdomain_data(sd)
+        p_name = self.pressure_variable
+        u_name = self.displacement_variable
+        t = self.time_manager.time
+
+        # Instantiate data class
+        out = SaveData()
+
+        # Store time data
+        out.time = t
+        out.nondim_time = self.nondim_time(t)
+
+        # Spatial variables
+        out.vertical_coo = sd.cell_centers[1]
+        out.nondim_vertical_coo = self.nondim_length(sd.cell_centers[1])
+
+        # Pressure data
+        out.numerical_pressure = data[pp.STATE][p_name]
+        out.exact_pressure = self.exact_sol.pressure(out.vertical_coo, out.time)
+        out.numerical_nondim_pressure = self.nondim_pressure(out.numerical_pressure)
+        out.exact_nondim_pressure = self.nondim_pressure(out.exact_pressure)
+
+        # Mechanical data
+        out.numerical_displacement = data[pp.STATE][u_name]
+        out.numerical_consolidation_degree = self.numerical_consolidation_degree(
+            out.numerical_displacement,  # displacement
+            out.numerical_pressure,  # pressure
+        )
+        out.exact_consolidation_degree = self.exact_sol.consolidation_degree(t)
+
+        # Store errors data
+        out.pressure_error = self.relative_l2_error(
+            sd,  # grid
```

Comment:
Here, I would prefer
```suggestion
            grid=sd,
```
etc.

Reply:
Yes, I agree. But for some reason, `mypy` complained when I wrote it like that. I will give it another go tomorrow.

Reply:
Apparently, there is an issue with data classes and Callable. I will investigate this further in the next PR. For the moment, I've added a #  type: ignore[call-arg]